src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserService.cs

```diff
@@ -132,90 +132,137 @@ public void InitializeService(ServiceHost serviceHost)
             FileBrowserCloseParams fileBrowserParams,
             RequestContext<FileBrowserCloseResponse> requestContext)
         {
-            FileBrowserCloseResponse response = new FileBrowserCloseResponse();
-            FileBrowserOperation removedOperation;
-            response.Succeeded = ownerToFileBrowserMap.TryRemove(fileBrowserParams.OwnerUri, out removedOperation);
-
-            if (removedOperation != null && this.expandNodeQueue != null)
+            try
             {
-                bool hasPendingQueueItems = this.expandNodeQueue.HasPendingQueueItems;
-                if (removedOperation.FileTreeCreated && !hasPendingQueueItems)
-                {
-                    removedOperation.Dispose();
-                    this.expandNodeQueue.CloseConnections(removedOperation.SqlConnection.DataSource, removedOperation.SqlConnection.Database, DefaultExpandTimeout);
-                }
-                else if (!removedOperation.FileTreeCreated)
-                {
-                    removedOperation.Cancel();
-                }
-                else if (hasPendingQueueItems)
-                { 
-                    this.expandNodeQueue.StopQueueProcessor(DefaultExpandTimeout);
-                }
+                var task = Task.Run(() => RunFileBrowserCloseTask(fileBrowserParams, requestContext))
```

Comment:
>ar task = [](start = 17, length = 9)

why are we saving into task variable?  it doesn't look like we use the variable.